src/main/java/chatbot/Alfred.java

```diff
@@ -93,120 +130,62 @@ public void printList() {
-        System.out.println("  " + list.get(extractedIdx));
-        System.out.println(line);
+        this.echo("Nice! I've marked this task as done:\n  " + list.get(extractedIdx).toString());
     }
 
     public void unmarkList(String index) {
-        int extractedIdx = 0;
-        try {
-        if (index.isEmpty()) {
-            throw new AlfredException("Sorry Master Bruce. Please enter a number after 'unmark'.");
-        }
-        try {
-            extractedIdx = Integer.parseInt(index);
-        }
-        catch (NumberFormatException e) {
-            if (list.size() == 1) {
-                throw new AlfredException("Sorry Master Bruce. Please enter a number. There is only 1 task in the list.");
-            } else {
-                throw new AlfredException("Sorry Master Bruce. Please enter a number in the range of 1 to " + list.size() + ".");
-            }
-        }
-        extractedIdx -= 1;
-        if (list.size() == 0) {
-            throw new AlfredException("Sorry Master Bruce. There are no tasks in the list.");
-        } else if (extractedIdx >= list.size() || extractedIdx < 0) {
-            if (list.size() == 1) {
-                throw new AlfredException("Sorry Master Bruce. The task number you have entered is not in the list. There is only 1 task in the list.");
-            } else {
-                throw new AlfredException("Sorry Master Bruce. The task number you have entered is not in the list. Please enter a number in the range of 1 to " + list.size() + ".");
-            }
-        } else if (!list.get(extractedIdx).isDone()) {
-            throw new AlfredException("Sorry Master Bruce. This task has already been marked as not done.");
+        int extractedIdx = validateAndExtractIndex(index);
+        if (extractedIdx == -1) {
+            return;
         }
-        } catch (AlfredException e) {
-            e.printError();
+        if (!list.get(extractedIdx).isDone()) {
+            this.echo("Sorry Master Bruce. This task has already been marked as not done.");
             return;
         }
         list.get(extractedIdx).toggleDone();
-        System.out.println(line);
-        System.out.println("OK, I've marked this task as not done yet:");
-        System.out.println("  " + list.get(extractedIdx));
-        System.out.println(line);
+        this.echo("OK, I've marked this task as not done yet:\n  " + list.get(extractedIdx).toString());
     }
 
     public void deleteList(String index) {
-        int extractedIdx = 0;
+        int extractedIdx = validateAndExtractIndex(index);
+        if (extractedIdx == -1) {
+            return;
+        }
+        String removedTask = list.get(extractedIdx).toString();
+        list.remove(extractedIdx);
+        this.echo("Noted. I've removed this task:\n  " + removedTask + "\nNow you have " + list.size() + " tasks in the list.");
+    }
+    private int validateAndExtractIndex(String index) {
```

Comment:
Maybe it is a good idea to put some of the codes in this function into another helper function? 🤔 too much ```try``` ```if``` stick together.

Reply:
validateAndExtractIndex is already a helper function